hax-driver.py

```diff
@@ -115,11 +115,39 @@ def shell(command, expect=0, cwd=None, env={}):
 if options.verify_extraction:
     shell(["make", "-C", "proofs/fstar/extraction/"])
 elif options.kyber_reference:
+    # Delete all extracted F* files
+    shell(["rm", "-f", "./proofs/fstar/extraction/Libcrux*"])
+    # Extract the platform crate
+    shell(
+        [
+            "cargo",
+            "hax",
+            "-C",
+            "-p",
+            "libcrux-platform",
+            ";",
+            "into",
+            "fstar",
+            "--interfaces",
+            "+**",
+        ],
+        cwd=".",
+        env=hax_env,
+    )
+    # Copy over the platform fsti
+    shell(
+        [
+            "cp",
+            "sys/platform/proofs/fstar/extraction/Libcrux_platform.fsti",
+            "proofs/fstar/extraction/",
+        ]
+    )
+    # Extract ml-kem from libcrux
     shell(
         cargo_hax_into
         + [
             "-i",
-            "-** +libcrux::kem::kyber::** {} -libcrux::digest::** -libcrux::**::types::index_impls::**".format(
+            "-** +libcrux::kem::kyber::** +libcrux_platform::** {} -libcrux::**::types::index_impls::**".format(
```

Comment:
I think we should just `hax into` `libcrux` and `libcrux-platform` in one go:
```python
shell(
        cargo_hax_into
        + [
            "-C", "-p", "libcrux", "-p", "libcrux-platform", ";",
            "-i",
            "-** +libcrux::kem::kyber::** +libcrux_platform::** {} -libcrux::**::types::index_impls::**".format(
                exclude_sha3_implementations
            ),
            "fstar",
            "--interfaces",
            "+* -libcrux::kem::kyber::types",
        ],
        cwd=".",
        env=hax_env,
    )
```

and then either copy over the fsti or just include then in the makefile.
Including in the makefile might actually be the best option.

Reply:
Well, if that would actually work, yes :) But this fails just like `--deps` (hacspec/hax#468).
But changing the makefile rather than copying the file over is a good idea!

Reply:
Oh, that's not great... https://github.com/hacspec/hax/pull/469 fixes it though.
Shall we just merge that PR in Hax and use that?

Ok, do you want me to do that in this PR?

Reply:
Yeah, can you do the makefile and hax-driver changes here? I'm looking at https://github.com/hacspec/hax/pull/469 then we can get both in.

Reply:
I pushed those changes.
https://github.com/hacspec/hax/pull/469 will be soon in, I stopped the CI for now, will re-trigger it once hax' PR is merged in main